instrumentation/runtime-telemetry/runtime-telemetry-java17/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java17/Java17RuntimeMetricsBuilder.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.java17;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.incubator.config.internal.InstrumentationConfig;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.internal.JmxRuntimeMetricsFactory;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+/** Builder for {@link Java17RuntimeMetrics}. */
+public final class Java17RuntimeMetricsBuilder {
+
+  private final OpenTelemetry openTelemetry;
+  // Visible for testing
+  final EnumMap<JfrFeature, Boolean> enabledFeatureMap;
+
+  private boolean disableJmx = false;
+  private boolean enableExperimentalJmxTelemetry = false;
+  private Consumer<Runnable> shutdownHook =
+      runnable -> {
+        Runtime.getRuntime()
+            .addShutdownHook(new Thread(runnable, "OpenTelemetry RuntimeMetricsShutdownHook"));
+      };
+
+  Java17RuntimeMetricsBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+    enabledFeatureMap = new EnumMap<>(JfrFeature.class);
+    for (JfrFeature feature : JfrFeature.values()) {
+      enabledFeatureMap.put(feature, feature.isDefaultEnabled());
+    }
```

Comment:
It method providing a map could be added in `JfrFeature`.

Reply:
this code is only moved - but feel free to create an issue